consensus/bor/finality/bor_verifier.go

```diff
@@ -135,6 +135,10 @@ func borVerify(ctx context.Context, config *config, start uint64, end uint64, ha
 
 	// fetch the end block hash
 	block, err := config.blockReader.BlockByNumber(context.Background(), roTx, end)
+	if block == nil {
```

Comment:
In case of `err != nil` and `block == nil` (which is highly possible), the error might be suppressed as we'll return from here. Can we move this after `err != nil` check?